### DISCUSSION TITLE
I'd like to understand why the Windows filesystem is mounted on WSL with noexec and nodev options.

### DISCUSSION THREAD

**Discussion by ishuah** [microsoft/WSL#9651] [Q&A] [answered]
_I'm asking this question because I'm curious and I want to understand WSL better._

My environment is Windows 11, using WSL 2 running Ubuntu. I'm learning container fundamentals and I've encountered an interesting error when running [debootstrap](https://manpages.debian.org/stretch/debootstrap/debootstrap.8.en.html).

When I run it in the mounted Windows filesystem (`/mnt/c/`):

```
user@laptop:/mnt/c/Users/user/workshop$ sudo debootstrap --arch amd64 jammy container_fs http://archive.ubuntu.com/ubuntu
```

I get the following error:

```
mknod: /mnt/c/Users/user/workshop/container_fs/test-dev-null: Operation not supported
E: Cannot install into target '/mnt/c/Users/user/workshop/container_fs' mounted with noexec or nodev
```

I tried to remount `/mnt/c/Users/user/workshop/container_fs` with `exec` and `dev` options.
```
user@laptop:/mnt/c/Users/user/workshop$ sudo mount -o remount,exec,dev /mnt/c/Users/user/workshop/container_fs
```

Output:
```
mount: /mnt/c/Users/user/workshop/container_fs: mount point not mounted or bad option.
```

It was at this point that I tried running debootstrap on `/home/user/` and it worked without any issues.

**Comment by NotTheDr01ds** [comment, marked as answer]
Answered in detail on your [Super User post](https://superuser.com/q/1769799/1210833).  Summary:

* 9P doesn't support `mknod`

**Reply by ishuah** [reply]
Thank you @NotTheDr01ds!
